fixtures/projection.go

```diff
@@ -0,0 +1,89 @@
+package fixtures
+
+import (
+	"context"
+	"time"
+
+	"github.com/dogmatiq/dogma"
+)
+
+// ProjectionMessageHandler is a test implementation of dogma.ProjectionMessageHandler.
+type ProjectionMessageHandler struct {
+	ConfigureFunc       func(dogma.ProjectionConfigurer)
+	HandleEventFunc     func(context.Context, []byte, []byte, []byte, dogma.ProjectionEventScope, dogma.Message) (bool, error)
+	ResourceVersionFunc func(context.Context, []byte) ([]byte, error)
+	CloseResourceFunc   func(context.Context, []byte) error
+	TimeoutHintFunc     func(m dogma.Message) time.Duration
+}
+
+var _ dogma.ProjectionMessageHandler = &ProjectionMessageHandler{}
+
+// Configure configures the behavior of the engine as it relates to this
+// handler.
+//
+// c provides access to the various configuration options, such as specifying
+// which types of event messages are routed to this handler.
+//
+// If h.ConfigureFunc is non-nil, it calls h.ConfigureFunc(c)
+func (h *ProjectionMessageHandler) Configure(c dogma.ProjectionConfigurer) {
+	if h.ConfigureFunc != nil {
+		h.ConfigureFunc(c)
+	}
+}
+
+// HandleEvent handles a domain event message that has been routed to this
+// handler.
+//
+// s provides access to the operations available within the scope of handling m.
+//
+// It panics with the UnexpectedMessage value if m is not one of the event
+// types that is routed to this handler via Configure().
+//
+// If h.HandleEventFunc is non-nil it calls h.HandleEventFunc(ctx, r,c,n, s, m).
```

Comment:
Any reason for `r,c,n,` to not have spaces between them?

Reply:
Also, do they need some basic docs to describe them?

Reply:
Hrm, it does have the basic docs from the interface it implements:

> // HandleEvent handles a domain event message that has been routed to this
> // handler.

I could copy the lot from the interface I suppose, but it just creates multiple places to update it. Each type clearly states which interface it mocks.

Reply:
Fixed spacing of variable names in the docs.

Reply:
When I looked at it, I couldn't remember what the `r`, `c` and `n` meant. They are all just `[]byte`. Its probably fine, I just thought I should mention it.

Reply:
You're not wrong - it's not great.  I just didn't want to reproduce the entire interface docs here as they're very long in this project. I guess this interface in particular is rather complex, I'll add some descriptions.

Reply:
Adding a couple of extra letters to those variable names might be enough, without adding any more doc text.